### DISCUSSION TITLE
Is there a way for recorder mode to capture and replay clicks under a dropdown menu

### DISCUSSION THREAD

**Discussion by rrajpuro** [seleniumbase/SeleniumBase#1902] [Q&A] [answered]
In the recorder mode, the links under a dropdown menu are not clickable during a replay as the recorder does not capture the hovering over a drop-down tab action. Is there any way that we can save our hover-over actions into the browser's sessionStorage? or is it unfeasible as there is no way to differentiate between a hover-over dropdown and just normal browsing?

P.S. I see that there is something about `# Handle a special case of links hidden in dropdowns` on line 373 in the base_case.py (seleniumbase 4.14.14) click function. But when I replay the session I get the Error `MyTestClass::test_base - selenium.common.exceptions.ElementNotVisibleException: Message`

**Comment by mdmintz** [comment, marked as answer]
For special cases, eg. on https://seleniumbase.io/demo_page, the Recorder will be able to detect hovering and clicking elements, as it did generate this script:

```python
from seleniumbase import BaseCase
BaseCase.main(__name__, __file__)

class RecorderTest(BaseCase):
    def test_recording(self):
        self.open("https://seleniumbase.io/demo_page")
        self.hover_and_click("div#myDropdown", "a#dropOption1")
        self.hover_and_click("div#myDropdown", "a#dropOption2")
        self.hover_and_click("div#myDropdown", "a#dropOption3")
```

However, for many situations, it won't differentiate between hovering and normal browsing, unless it can detect a hover_click action like above, but even that requires a straightforward HTML chunk that makes it clear enough to detect the hovering and clicking.

If you replace those ``self.click(selector)`` actions with ``self.js_click(selector)`` actions, then your script will be able to click on hidden elements that are located somewhere in the HTML, even if not directly visible without hovering over another element first.
